src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java

```diff
@@ -0,0 +1,167 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+import org.zalando.problem.Problem;
+
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static javax.ws.rs.core.Response.Status.PRECONDITION_FAILED;
+import static javax.ws.rs.core.Response.Status.SERVICE_UNAVAILABLE;
+import static org.zalando.problem.MoreStatus.UNPROCESSABLE_ENTITY;
+
+@RestController
+public class EventStreamController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EventStreamController.class);
+
+    private final TopicRepository topicRepository;
+
+    private final ObjectMapper jsonMapper;
+
+    public EventStreamController(final TopicRepository topicRepository, final ObjectMapper jsonMapper) {
+        this.topicRepository = topicRepository;
+        this.jsonMapper = jsonMapper;
+    }
+
+    @Timed(name = "stream_events_for_event_type", absolute = true)
+    @RequestMapping(value = "/event-types/{name}/events", method = RequestMethod.GET)
+    public StreamingResponseBody streamEventsFromPartition(
+            @PathVariable("name") final String eventTypeName,
+            @RequestParam(value = "batch_limit", required = false, defaultValue = "1") final int batchLimit,
+            @RequestParam(value = "stream_limit", required = false, defaultValue = "0") final int streamLimit,
+            @RequestParam(value = "batch_flush_timeout", required = false, defaultValue = "30") final int batchTimeout,
+            @RequestParam(value = "stream_timeout", required = false, defaultValue = "0") final int streamTimeout,
+            @RequestParam(value = "stream_keep_alive_limit", required = false, defaultValue = "0") final int streamKeepAliveLimit,
+            @Nullable @RequestHeader(name = "X-nakadi-cursors", required = false) final String cursorsStr,
+            final NativeWebRequest request, final HttpServletResponse response) throws IOException {
+
+        return outputStream -> {
+            try {
+                // todo: we should get topic from EventType after persistence of EventType is implemented
```

Comment:
That's an open question. I like the idea of using event type name, since it's immutable and enforces the 1:1 parity between kafka topics and event types :+1: